agent/stats/queue.go

```diff
@@ -137,6 +138,101 @@ func (queue *Queue) GetStorageWriteStatsSet() (*ecstcs.ULongStatsSet, error) {
 	return queue.getULongStatsSet(getStorageWriteBytes)
 }
 
+// GetNetworkStatsSet gets the stats set for network metrics.
+func (queue *Queue) GetNetworkStatsSet() (*ecstcs.NetworkStatsSet, error) {
+	networkStatsSet := &ecstcs.NetworkStatsSet{}
+	var err error
+	networkStatsSet.RxBytes, err = queue.getULongStatsSet(getNetworkRxBytes)
+	if err != nil {
+		seelog.Warnf("Error getting network rx bytes: %v", err)
+	}
+	networkStatsSet.RxDropped, err = queue.getULongStatsSet(getNetworkRxDropped)
+	if err != nil {
+		seelog.Warnf("Error getting network rx dropped: %v", err)
+	}
+	networkStatsSet.RxErrors, err = queue.getULongStatsSet(getNetworkRxErrors)
+	if err != nil {
+		seelog.Warnf("Error getting network rx errors: %v", err)
+	}
+	networkStatsSet.RxPackets, err = queue.getULongStatsSet(getNetworkRxPackets)
+	if err != nil {
+		seelog.Warnf("Error getting network rx packets: %v", err)
+	}
+	networkStatsSet.TxBytes, err = queue.getULongStatsSet(getNetworkTxBytes)
+	if err != nil {
+		seelog.Warnf("Error getting network tx bytes: %v", err)
+	}
+	networkStatsSet.TxDropped, err = queue.getULongStatsSet(getNetworkTxDropped)
+	if err != nil {
+		seelog.Warnf("Error getting network tx dropped: %v", err)
+	}
+	networkStatsSet.TxErrors, err = queue.getULongStatsSet(getNetworkTxErrors)
+	if err != nil {
+		seelog.Warnf("Error getting network tx errors: %v", err)
+	}
+	networkStatsSet.TxPackets, err = queue.getULongStatsSet(getNetworkTxPackets)
+	if err != nil {
+		seelog.Warnf("Error getting network tx packets: %v", err)
+	}
+	return networkStatsSet, err
+}
+
+func getNetworkRxBytes(s *UsageStats) uint64 {
+	if s.NetworkStats != nil {
+		return s.NetworkStats.rxBytes
+	}
+	return uint64(0)
```

Comment:
for this and subsequent stats where do we surface that our s.<stat> is nil?  
This looks like a silent failure.

Reply:
this is not exactly a silent failure, since some network modes might have this as nil. hence just sending 0 . what do you think?

Reply:
This is fine, as long as we are capturing true stats failures elsewhere.  We will need to clearly define what our true failure cases are -- for example, for storage stats, I'd assume any null stats represent a true failure and I need to surface that somehow.  I might return a uint64(0) here for storage stats, but I need to be sure that I've got a debug statement linked to this failure that will help me troubleshoot.

Reply:
If you agree that we need this, can we at least add a TODO here to be sure we don't lose the thought?

Reply:
network stats being nil is a valid case for host/awsvpc mode. 
instead of throwing an error, I have added a debug statement here https://github.com/aws/amazon-ecs-agent/pull/2016/files#diff-598be0dc5499d16fd579d5ce5dee96abR57